### PR TITLE
test-suite: fix bug causing unit tests to be skipped

### PR DESCRIPTION
**Pull request by SkySkimmer**
Since `ocaml_pwd.ml` was added this unquoted glob would be expanded by
the shell before being passed to `find`.

### CODE REVIEW

**Review by herbelin — APPROVED** [review]
I could not get the reason why `ocaml_pwd.ml` changed the effect of `find` but the fix is obviously correct.

As a conclusion, `proof_diffs_test` has not been untested for long, right? Do you have an idea of how it is supposed to be fixed?

**Comment by SkySkimmer** [comment]
The `lex_tests.ml` never worked, it was skipped by the bug this PR fixes. I'm reverting its addition, it seems easier this way.
@gares you may want to make a PR adding it back fixed once this PR is merged.

**Comment by SkySkimmer** [comment]
@herbelin are you assigning?

**Comment by SkySkimmer** [comment]
BTW we may want to fix this in the 8.10 branch too.

**Comment by herbelin** [comment]
> BTW we may want to fix this in the 8.10 branch too.

Why not!

> @herbelin are you assigning?

I do. Waiting for some acknowledgement from @gares about `lex_tests.ml` and merging tomorrow late in any cases.

**Comment by JasonGross** [comment]
> I could not get the reason why `ocaml_pwd.ml` changed the effect of `find` but the fix is obviously correct.

The reason is that if there is any file in the current directory matching the glob `*.ml`, then the shell will expand it to match all such files.  If there are no files matching the glob, then it will leave it as `*.ml`.  Hence, before `ocaml_pwd.ml` was added, there were no `.ml` files in the relevant directory, and so the shell passed `*.ml` through unchanged.  But once it was added, the shell expanded `*.ml` to `ocaml_pwd.ml`.  (The lesson I'm going to claim from this is that things should always be quoted in bash, even when they don't need to be.)

**Comment by herbelin** [comment]
@JasonGross: Ah, ok, thanks!

**Comment by SkySkimmer** [comment]
Got
~~~
File "./primitive/float/compare.v", line 324, characters 0-50:
Error:
Dynlink error: The module `Coq_native6feb64' is already loaded (either by the main program or a previously-dynlinked library)
~~~
from test-suite:edge+flambda

I can't see how that could be related, so I guess we can merge.

**Review by JasonGross — DISMISSED** [review]
I don't think we can merge yet; the Mac OS job fails with
```
TEST      unit-tests/clib/unicode_tests.ml
ocamlfind: Package `coq.toplevel' not found
make[2]: *** [unit-tests/clib/unicode_tests.ml.log] Error 2
make[2]: *** Waiting for unfinished jobs....
TEST      unit-tests/clib/inteq.ml
ocamlfind: Package `coq.toplevel' not found
make[2]: *** [unit-tests/clib/inteq.ml.log] Error 2
make[1]: *** [test-suite] Error 2
make: *** [submake] Error 2
```
see https://dev.azure.com/coq/coq/_build/results?buildId=9682&view=logs&j=a5e52b91-c83f-5429-4a68-c246fc63a4f7&t=bde16fc7-7659-5b1c-12aa-0110866d993c , https://github.com/coq/coq/pull/12289/checks?check_run_id=661320496

**Comment by JasonGross** [comment]
I'm going to restart [the failing gitlab job](https://gitlab.com/coq/coq/-/jobs/545904250) to see if it's reproducible

**Comment by ejgallego** [comment]
> I can't see how that could be related, so I guess we can merge.

That looks kinda worrying [as discussed on Zulip] , seems rare but it could be some kind of race on the native loader.

**Comment by ejgallego** [comment]
Adding a `needs:fixing` as to see if we can fix the OSX issue. Likely the azure job is missing this line from .gitlab-ci.yml:

```
    - export OCAMLPATH=$(readlink -f ../_install_ci/lib/):"$OCAMLPATH"
```

**Comment by ejgallego** [comment]
I've pushed a hotfix, let's see.

**Comment by SkySkimmer** [comment]
Looks like it works.

Not sure what's going on with `test-suite:base+async`, it's probably not related though.

File: azure-pipelines.yml [hunk 1]
```diff
@@ -97,7 +98,10 @@ jobs:
 
   - script: |
       eval $(opam env)
+      export OCAMLPATH=$(AZURE_BUILDIR):"$OCAMLPATH"
```
Comment (SkySkimmer) [inline azure-pipelines.yml:101]:
```suggestion
      export OCAMLPATH=$AZURE_BUILDIR:"$OCAMLPATH"
```
or
```suggestion
      export OCAMLPATH=$(Agent.BuildDirectory):"$OCAMLPATH"
```
(without an `env` declaration, same syntax as the -prefix above)

**Comment by ejgallego** [comment]
> Not sure what's going on with `test-suite:base+async`, it's probably not related though.

It has been broken since a while, likely due to modification of the error printing attaching more info to exceptions?

**Comment by ejgallego** [comment]
Bah this still doesn't work, what a pain in the ass.... Let me push a debug commit [to be removed]

**Comment by ejgallego** [comment]
Here's the problem:
```
Agent.BuildDirectory: /Users/runner/runners/2.168.2/work/1
pwd: /Users/runner/runners/2.168.2/work/1/s
```

**Comment by SkySkimmer** [comment]
Let's try using pwd then

**Comment by SkySkimmer** [comment]
It passed, I squashed.

**Comment by ejgallego** [comment]
@herbelin you can merge.

**Comment by herbelin** [comment]
> @herbelin you can merge.

Is it still relevant now?

**Comment by ejgallego** [comment]
> Is it still relevant now?

Yes, this needs to be merged.

**Comment by ejgallego** [comment]
Thanks @herbelin

**Pull request merged**